packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -191,3 +190,67 @@ function isSecondsPropertyMisusedInCalendarTriggerInput(
       Object.keys(triggerWithoutChannelId).length > 1)
   );
 }
+
+function validateDateComponentsInTrigger(
+  trigger: NonNullable<NotificationTriggerInput>,
+  components: string[]
+) {
+  const anyTriggerType = trigger as any;
```

Comment:
yeah, shame, but I don't think I can avoid this unless I use way more complicated typings and overrides that will be very cryptic (I tried… and all the possibilities I found were not worth in my opinion). The problem is that I need to check (`in`) and access properties that the union type will not allow me (`in` only works with `any` or `object`, and then to access the property we need to have `any` or a type that actually has it).